flink-clients/src/main/java/org/apache/flink/client/cli/AbstractCustomCommandLine.java

```diff
@@ -91,4 +91,9 @@ protected Configuration applyCommandLineOptionsToConfiguration(CommandLine comma
 
 		return resultingConfiguration;
 	}
+
+	@Override
+	public boolean isDetachedMode(CommandLine commandLine) {
```

Comment:
I think the `CustomCommandLine` should not have this method. It is more of a factory for a `ClusterDescriptor` and general options like this should go to the `CLI` class.